src/alpaka/AlpakaCore/HistoContainer.h

```diff
@@ -0,0 +1,370 @@
+            assert((*off) > 0);
+            int32_t ih = off - offsets - 1;
+            assert(ih >= 0);
+            assert(ih < int(nh));
+            h->fill(acc, v[i], i, ih);
+          }
+          endElementIdx += gridDimension;
+        }
+      }
+    };
+
+    struct launchZero {
+      template <typename T_Acc, typename Histo>
+      ALPAKA_FN_ACC ALPAKA_FN_INLINE __attribute__((always_inline)) void operator()(const T_Acc &acc,
+                                                                                    Histo *__restrict__ h) const {
+        const auto &[firstElementIdxGlobal, endElementIdxGlobal] =
+            cms::alpakatools::element_global_index_range(acc, Vec1::all(Histo::totbins()));
+
+        for (uint32_t i = firstElementIdxGlobal[0u]; i < endElementIdxGlobal[0u]; ++i) {
+          h->off[i] = 0;
+        }
+      }
+    };
+
+    struct storePrefixScanWorkingSpace {
+      template <typename T_Acc, typename Histo>
+      ALPAKA_FN_ACC void operator()(const T_Acc &acc, Histo *__restrict__ h, const unsigned int nblocks) const {
+        h->psws = nblocks;
+      }
+    };
+
+    template <typename Histo>
+    ALPAKA_FN_HOST ALPAKA_FN_INLINE __attribute__((always_inline)) void launchFinalize(Histo *__restrict__ h,
+                                                                                       const DevAcc1 &device,
+                                                                                       Queue &queue) {
+      uint32_t *poff = (uint32_t *)((char *)(h) + offsetof(Histo, off));
+
+      const int num_items = Histo::totbins();
+
+      auto psum_dBuf = alpaka::mem::buf::alloc<uint32_t, Idx>(device, Vec1::all(num_items));
+      uint32_t *psum_d = alpaka::mem::view::getPtrNative(psum_dBuf);
+
+      const unsigned int nthreads = 1024;
+      const Vec1 threadsPerBlockOrElementsPerThread(nthreads);
+      const unsigned int nblocks = (num_items + nthreads - 1) / nthreads;
+      const Vec1 blocksPerGrid(nblocks);
+
+      alpaka::queue::enqueue(
+          queue,
+          alpaka::kernel::createTaskKernel<Acc1>(
+              WorkDiv1{Vec1::all(1u), Vec1::all(1u), Vec1::all(1u)}, storePrefixScanWorkingSpace(), h, nblocks));
+
+      const WorkDiv1 &workDiv = cms::alpakatools::make_workdiv(blocksPerGrid, threadsPerBlockOrElementsPerThread);
+      alpaka::queue::enqueue(queue,
+                             alpaka::kernel::createTaskKernel<Acc1>(
+                                 workDiv, multiBlockPrefixScanFirstStep<uint32_t>(), poff, poff, psum_d, num_items));
+
+      const WorkDiv1 &workDivWith1Block =
+          cms::alpakatools::make_workdiv(Vec1::all(1), threadsPerBlockOrElementsPerThread);
+      alpaka::queue::enqueue(
+          queue,
+          alpaka::kernel::createTaskKernel<Acc1>(
+              workDivWith1Block, multiBlockPrefixScanSecondStep<uint32_t>(), poff, poff, psum_d, num_items, nblocks));
+      alpaka::wait::wait(queue);
```

Comment:
Yes that was a painful segfault to fix. Basically here, workdiv and kernels are defined in host function `launchFinalize` scope.
If that function returns before the device work is completed, we run into segfaults / spurious results.

Reply:
I presume a clean way would be to only have one host function instead of intricated host functions in which device work is defined.
But since here this is just a test where we do not care about perf, and want to keep the code similar to the CUDA versions, I just added those.

Reply:
Sounds strange. I would have naively expected `alpaka::queue::enqueue()` etc to copy the work division. Or could it be the allocated buffer that must stay alive until the work finishes (that I believe)?

Reply:
By quick look through Alpaka code they seem to copy both the work division and the functor within the call `alpaka::kernel::createTaskKernel()` call. This would need to be investigated further (because I'd really want these `wait()`s to go), but that can be done after this PR.

Reply:
>Sounds strange. I would have naively expected alpaka::queue::enqueue() etc to copy the work division. Or could it be the allocated buffer that must stay alive until the work finishes (that I believe)?

Yes, so looking at the execution task class (here for example with GPU CUDA backend):
http://alpaka-group.github.io/alpaka/TaskKernelGpuUniformCudaHipRt_8hpp_source.html#l00134 
- the kernel function object and its arguments, are move / copy constructed and stored as TaskKernel data members.
- the workdiv is also move / copy constructed, and stored as data member of the base class (TaskKernel is derived from WorkDiv). http://alpaka-group.github.io/alpaka/WorkDivMembers_8hpp_source.html#l00052

Now, a raw pointer is passed as argument (`poff`), and was obtained from the Alpaka-equivalent to the get() (getPtrNative) from a reference-counting buffer handle.
So, if the handle is reference-counted to 0 before the device work finishes, we end up with a dangling pointer, and we are still screwed.
However, that is not the case, because the owning pointer is `h_d`, and is defined out of the scope of this function.

So evth should be fine.

Trying to run without the `alpaka::wait::wait(queue)`, I still sometimes get a segfault with `./HistoContainer_t.tbb`. It is a spurious segfault, which makes it hard to know whether sth fixes it or not. It seems to be somehow stemming from a synchronization issue indeed, but the issue is not compulsory (should not) be here.
Hence will remove the `alpaka::wait::wait(queue)`.